tests/test_doc_auto_generation.py

```diff
@@ -0,0 +1,87 @@
+from docs import autogen
+import pytest
+
+test_doc1 = {
+    'doc': '''This function does something really interesting.
```

Comment:
Please use a real docstring from the codebase (if necessary modify it to include all features being tested)